src/config/parser.rs

```diff
@@ -118,6 +122,15 @@ impl Config {
             _ => parsed_cet,
         };
 
+        let proxy_str = globals.get::<_, String>("proxy")?;
+        let proxy = match Proxy::all(proxy_str) {
+            Ok(proxy) => Some(proxy),
+            Err(_) => {
+                log::error!("Invalid proxy url, defaulting to no proxy.");
+                None
+            }
+        };
+
```

Comment:
```suggestion
                let proxy_opt = globals.get::<_, Option<String>>("proxy")?;
        let proxy = proxy_opt.and_then(|proxy_str| {
            Proxy::all(proxy_str).ok().and_then(|_| {
                log::error!("Invalid proxy url, defaulting to no proxy.");
                None
            })

```

I would suggest having the following parsing code which is more easy to read as well it uses `Option<String>` to parse the option from the config. So that we can have the following way for the user to disable or enable the option which I feel better conveys and documents like whether the option is disabled or not. While for any other unexpected cases like empty string or invalid proxy url can be handled via the above code with `Proxy::all`. 

**Disabled:**
```Lua
proxy = nil
```

**Enabled:**
```Lua
proxy = "socks5h://127.0.0.1:9050"
```